### PR TITLE
added node methods for cid and tid requests returning promises and th…

### PR DESCRIPTION
**Pull request by grantglidewell**
added node methods for cid and tid requests returning promises and then mapping once resolved.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -25,7 +25,44 @@ module.exports = class {
    */
   issueRequest(method, url) {
     return this.axios({ method, url })
-      .then(({ data }) => data);
+      .then(({ data }) => data)
+      .catch(Promise.reject);
```
Comment (mattgrill) [inline lib/index.js]: Lets not try and figure out how to handle errors in this PR.

File: lib/index.js [hunk 2]
```diff
@@ -40,6 +77,30 @@ module.exports = class {
...
+      `${this.base}node.${this.format}?${qs.stringify(filterOptions, { indices: false })}`);
+  }
+  /**
+   * Node Taxonomy Id
+   * @param  {Object} [filterOptions={}]
+   *   Query string filter options
+   * @return {Promise}
+   *   Pending issueRequest promise.
+   */
+  nodeTaxonomyId(filterOptions = {}) {
```
Comment (mattgrill) [inline lib/index.js]: Lets normalize the method name, perhaps something like `getRelatedTaxonomyTerms`?

File: lib/index.js [hunk 3]
```diff
@@ -25,7 +25,44 @@ module.exports = class {
...
+   * tid Request wrapper.
+   * @param  {String} method
+   *   Method for the HTTP request.
+   * @param  {String} url
+   *   The URL for the HTTP request.
+   * @return Mapped {Promises}
+   *   The response from the Drupal.org HTTP api.
+   */
+  tidRequest(method, url) {
+    return this.axios({ method, url })
```
Comment (mattgrill) [inline lib/index.js]: We already have a request handler that should be used. Lets not add another reference to `axios` and all that comes with that.

**Review by mattgrill — CHANGES_REQUESTED** [review]
Couple of other non-specific issues. 

`getRelatedTaxonomyTerms` should be called after fetching the node data, which contains the references to the related taxonomy terms.

`taxonomyTerm` returns a `Promise` so we can construct an array of requests, `Promise.all()` to fetch all the related terms and then handle the result.

**Comment by grantglidewell** [comment]
New commit covers all issues mentioned

File: lib/index.js [hunk 4]
```diff
@@ -112,4 +112,33 @@ module.exports = class {
...
+  /**
+   * getRelatedTaxonomyTerms
+   * Related Taxonomy Term Request wrapper.
+   * @param  {Promise} data
+   *   Taxonomy IDs from a node query.
+   * @return {Promise}
+   *   The response from the Drupal.org HTTP api.
+   */
+  getRelatedTaxonomyTerms(data) {
+    const taxonomyTermsArray = data.list[0].taxonomy_vocabulary_9;
```
Comment (mattgrill) [inline lib/index.js]: I think we had talked about only accepting an array of `nid`'s? That way we don't have to even think about the format of the data, and then just check that the length of the array passed is greater than `0` and some basic sanity checks around the contents.

**Review by mattgrill — CHANGES_REQUESTED** [review]
Looks like you accidentally included the `package-lock.json` from installing dependencies from with npm5.

**Review by mattgrill — APPROVED** [review]

**Pull request merged**